agent/src/jmh/java/com/code_intelligence/jazzer/instrumentor/EdgeCoverageInstrumentation.java

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.instrumentor;
+
+import static java.lang.String.format;
+
+import com.code_intelligence.jazzer.MockDriver;
+import com.code_intelligence.jazzer.runtime.CoverageMap;
+import com.github.fmeum.rules_jni.RulesJni;
+import java.io.*;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.openjdk.jmh.annotations.*;
+
+@Warmup(iterations = 10, time = 3)
+@Measurement(iterations = 10, time = 3)
+@Fork(value = 3)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@BenchmarkMode(Mode.AverageTime)
+@State(Scope.Benchmark)
+@SuppressWarnings("unused")
+public class EdgeCoverageInstrumentation {
+  private Object exampleObject;
+  private Method exampleMethod;
```

Comment:
I would prefer `MethodHandle` over `Method` in benchmarks as the reflection overhead may hide performance differences.

Reply:
Done. I tried to use `LambdaMetafactory` as that should be even faster but had no luck. 
Currently, I don't want to spend more time on this but probably will get back to it :-)

Reply:
I want to learn more about `LambaMetaFactory` anyway, let's make that a Friday topic at some point.